### ISSUE TITLE
False positive in `unconditional-recursion` rule

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#596]
**Describe the bug**
The rule `unconditional-recursion` produce false positives in presence of anonymous receivers.
For example, for the following code
```go
func (*InternalMessageInfo) DiscardUnknown(m Message) {
        DiscardUnknown(m)
}
```
it will warn about an unconditional recursive call.

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
revive  -config rec.toml github.com/golang/protobuf/proto/deprecated.go
```

```toml
# config file
ignoreGeneratedHeader = false
severity = "warning"
confidence = 0.5
errorCode = 0
warningCode = 0

[rule.unconditional-recursion]
```

Ouput:
```
github.com/golang/protobuf/proto/deprecated.go:92:2: unconditional recursive call
```

**Expected behavior**
No failures
